### DISCUSSION TITLE
OIDC Auth Loop + Role Assignment via Azure/OIDC

### DISCUSSION THREAD

**Discussion by crsnchng** [Budibase/budibase#5218] [Help] [answered]
Hello!

I may just have a fundamental misunderstanding of how OIDC works however I've run into some issues during testing. 

Auth using Azure and OIDC seems to work however when a user in my Azure tenant logs into Budibase for the first time, they get stuck in a loop in the web browser. This loop is resolved by assigning the user (which was auto created in the Budibase user list upon login) the correct permissions to view the page they were trying to view.

It seems that if a role can be assigned during the initial login process, the loop issue will be solved however I can't seem to figure out how this would work via Azure or where this would be set up.

Wondering if anyone else has this figured out or if someone could yell at me for doing it wrong and let me know what the correct method would be.

Thanks in advance.

**Comment by melohagan** [comment, marked as answer]
Hello!

This issue was raised here: https://github.com/Budibase/budibase/issues/5004

It should now be fixed in todays release!

**Reply by crsnchng** [reply]
I see it now throws an insufficient permissions page - this is great, Do you by chance know if there is a way to assign default roles somehow?

**Reply by melohagan** [reply]
There's no way to do default roles at the moment. You'll have to manually give users access to apps for now.
